fatiando/gridder.py

```diff
@@ -394,3 +395,55 @@ def cut(x, y, scalars, area):
               if x[i] >= xmin and x[i] <= xmax
               and y[i] >= ymin and y[i] <= ymax]
     return [x[inside], y[inside], [s[inside] for s in scalars]]
+
+
+def cut_regular(x, y, scalars, shape, area):
+    """
+    Return a subsection of a regular grid.
+
+    The returned subsection is not a copy! In technical terms, returns a slice
+    of the numpy arrays. So changes made to the subsection reflect on the
+    original grid. Use numpy.copy to make copies of the subsections and avoid
+    this.
+
+    Parameters:
+
+    * x, y
+        Arrays with the x and y coordinates of the data points.
+    * scalars
+        List of arrays with the scalar values assigned to the grid points.
+    * shape: tuple(ny,nx)
+        Shape of the original data points.
+    * area
+        ``(x1, x2, y1, y2)``: Borders of the subsection
+
+    Returns:
+
+    * ``[subx, suby, subscalars,shape_window]``
+        Arrays with x and y coordinates and scalar values of the subsection.
+    shape_window: tuple(ny,nx)
+        Shape of the cutted grid.
+
+    """
+    xmin, xmax, ymin, ymax = area
+    if xmin > xmax:
+        xmin, xmax = xmax, xmin
+    if ymin > ymax:
+        ymin, ymax = ymax, ymin
+
+    # Calculates the distance between neighbour points
+    if shape[0]*shape[1] != len(x):
+        raise ValueError("The shape isn't the shape of the x,y grid.")
+
+    distance_x = abs(max(x)-min(x)) / (shape[1]-1)
+    distance_y = abs(max(y)-min(y)) / (shape[0]-1)
+    shape_cutted = (int(abs(ymax-ymin)/distance_y + 1),
+                    int(abs(xmax-xmin)/distance_x + 1))
+
+    if len(x) != len(y):
+        raise ValueError("x and y must have the same length")
+    inside = [i for i in xrange(len(x))
+              if x[i] >= xmin and x[i] <= xmax
+              and y[i] >= ymin and y[i] <= ymax]
```

Comment:
A nice thing about dealing with regular grids is that you don't have to do these calculations. Calculating `inside` can be time consuming for large grids and is not really straight forward. This is something that has to be done for irregular data, but grid data offer a better way.

If you `reshape` the input arrays into 2D arrays (`xg = x.reshape(shape)`), then you can use numpy slicing to get the subgrid. First you'll need to calculate the max and min i, j indexes of the subgrid (using the spacing between points). Once you have imin, imax, jmin, jmax, getting the subgrid becomes `subx = xg[imin:imax, jmin:jmax]`. The shape of the subgrid is `subx.shape` and you can return 1D arrays with `subx.ravel()`.